api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto

```diff
@@ -146,13 +147,28 @@ message TcpProxy {
   // The router selects an upstream cluster based on these weights.
   message WeightedCluster {
     message ClusterWeight {
-      // Name of the upstream cluster.
+      // Name of the upstream cluster. The cluster must exist in the
+      // :ref:`cluster manager configuration <config_cluster_manager>`.
       string name = 1 [(validate.rules).string.min_bytes = 1];
 
       // When a request matches the route, the choice of an upstream cluster is
       // determined by its weight. The sum of weights across all entries in the
       // clusters array determines the total weight.
       uint32 weight = 2 [(validate.rules).uint32.gte = 1];
+
+      // [#not-implemented-hide:] Optional endpoint metadata match
+      // criteria. Only endpoints in the upstream cluster with metadata
+      // matching that set in metadata_match will be considered. The filter
+      // name should be specified as *envoy.lb*.
+      envoy.api.v2.core.Metadata metadata_match = 3;
```

Comment:
What's the rationale for adding these two new fields in this PR? They're not implemented, so unless there's a good reason, let's wait to add them until they're implemented.

Reply:
Reverted in 82e4765.